### PR TITLE
feat: Add Boardsouce Unicorne

### PR DESCRIPTION
**Pull request by kdb424**
Will come back and offer a follow up PR for images and buy links when there is a store page and official photos, but some are in the wild already, hence the "early" PR.

### CODE REVIEW

File: boards/boardsource/unicorne/kb.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    SDA=board.SDA
+    brightness_limit = 0.0
+    num_pixels = 70
+    led_key_pos = [24,23,18,17,10,9,36,37,44,45,50,51,
+                25,22,19,16,11,8,35,38,43,46,49,52,
+                26,21,20,15,12,7,34,39,42,47,48,53,
+                14,13,6,33,40,41,
+                2,1,0,27,28,29,3,4,5,32,31,30]
+    brightness_limit = 0.5
+    num_pixels = 54
```
Comment (xs5871) [inline boards/boardsource/unicorne/kb.py:38]: Keyboard class has duplicate attributes.

File: boards/boardsource/unicorne/main.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+OS_LCTL = KC.OS(KC.LCTL)
+OS_LSFT = KC.OS(KC.LSFT)
+OS_LALT = KC.OS(KC.LALT)
+keyboard.modules.append(oneshot)
+keyboard.modules.append(layers_ext)
+keyboard.modules.append(modtap)
+keyboard.modules.append(combos)
+
+# oled
+oled_ext = Oled(
```
Comment (xs5871) [inline boards/boardsource/unicorne/main.py:40]:
```suggestion
oled = Oled(
```

File: boards/boardsource/unicorne/main.py [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+            1: ['', '', '', '', '', '', ' ^', '   ^'],
+        },
+    ),
+    toDisplay=OledDisplayMode.TXT,
+    flip=True,
+)
+# oled
+keyboard.extensions.append(oled_ext)
+# ledmap
+rgb_ext = Rgb_matrix(
```
Comment (xs5871) [inline boards/boardsource/unicorne/main.py:65]:
```suggestion
rgb = Rgb_matrix(
```

File: boards/boardsource/unicorne/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Unicorne
+
+Based on the legendary corne (crkbd) from Foostan with more features than ever before.
+
+- Prebuilt at the factory
+- Hotswap sockets
+- RP2040 with 16MB of ROM for tons of room for fun things!
+- Oled screes
```
Comment (xs5871) [inline boards/boardsource/unicorne/README.md:8]:
```suggestion
- OLED screens
```

**Review by xs5871 — CHANGES_REQUESTED** [review]

File: boards/boardsource/unicorne/kb.py (hunk 1 above)
Comment (kdb424) [reply boards/boardsource/unicorne/kb.py]: Think this is resolved now.

**Comment by kdb424** [comment]
Not sure how to fix that isort. My local system seems to say it's correct, and runs without issue. Other PR also cleans up isort that my system kept formatting. Can't run `make fix-formatting` or `make test` on OSX or any of my Linux systems within the virtualenv as it seems to also be missing deps. If you want to correct this somehow, that would be appreciated, but I'm unable to figure out how to format the file in a way that the CI is happy with working blind.

**Review by xs5871 — APPROVED** [review]

**Pull request merged**